packages/react/ds/src/side-nav/side-nav.tsx

```diff
@@ -17,173 +25,228 @@ const SideNavContext = React.createContext<SideNavContextType | undefined>(
   undefined,
 );
 
+const ItemContent = memo(
+  ({
+    icon,
+    label,
+    showExpandableIcon,
+    isOpen,
+  }: {
+    icon?: IconId;
+    label: string;
+    showExpandableIcon?: boolean;
+    isOpen?: boolean;
+  }) => {
+    return (
+      <>
+        <div className="gi-side-nav-item-left">
+          {icon && (
+            <div className="gi-side-nav-item-icon">
+              <Icon icon={icon} />
+            </div>
+          )}
+          <div className="gi-side-nav-item-label">
+            <Paragraph size="md">{label}</Paragraph>
+          </div>
+        </div>
+        {showExpandableIcon && (
+          <div className="gi-side-nav-expandable-icon">
+            <Icon
+              className={cn(isOpen && 'gi-rotate-180')}
+              icon="keyboard_arrow_down"
+            />
+          </div>
+        )}
+      </>
+    );
+  },
+);
+
 export const SideNavItem: React.FC<
   PropsWithChildren<SideNavItemProps> & { open?: boolean }
-> = ({
-  children,
-  parent,
-  expandable,
-  label,
-  value,
-  icon,
-  href,
-  open = false,
-}) => {
-  const context = React.useContext(SideNavContext);
-
-  if (!context) {
-    throw new Error('SideNavItem must be used within a SideNav');
-  }
-
-  const {
-    openItemIds,
-    selectedItemId,
-    setOpenItemIds,
-    setSelectedItemId,
-    navId,
-  } = context;
-
-  const isOpen = openItemIds.includes(value);
-  const isSelected = selectedItemId === value;
-
-  useEffect(() => {
-    if (open) {
-      setOpenItemIds((previousIds: string[]) => {
-        if (!previousIds.includes(value)) {
-          return [...previousIds, value];
-        }
-        return previousIds;
-      });
+> = React.memo(
+  ({
+    children,
+    primary,
+    secondary,
+    expandable,
+    label,
+    value,
+    icon,
+    href,
+    hrefComponent,
```

Comment:
Check what has been done on the Breadcrumb.
The `children` of the `SideNavItem` could be a string or a component and `asChild` is a prop.